src/OptionSettlementEngine.sol

```diff
@@ -800,13 +805,16 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
                 unexercisedBucketIndices[exerciseIndex] = overwrite;
                 unexercisedBucketIndices.pop();
             } else {
-                // Bucket is partially exercised/assigned
+                // Bucket is partially exercised/assigned.
                 amountPresentlyExercised = amount;
                 amount = 0;
             }
             bucketInfo.amountExercised += amountPresentlyExercised;
 
+            emit BucketAssignedExercise(optionId, bucketIndex, amountPresentlyExercised);
```

Comment:
I believe `amountPresentlyExercised` is the correct argument to include here — the amount of options that were just assigned exercised from this bucket.